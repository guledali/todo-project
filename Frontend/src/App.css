.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
}

.App-header {
  /* background-color: #282c34; */
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.screen {
  height: 100vh;
  width: 100vw;
  background: #f8f9fa;
}

.h-10 {
  height: 10%;
}
.pt-10 {
  padding-top: 5rem;
}

.custom-size {
  font-size: 2rem;
  font-weight: medium;
}
.m25 {
  margin-top: 100%;
}