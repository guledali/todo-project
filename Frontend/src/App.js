import React, { Component } from 'react';

import './App.css';
import Form from './components/Form';
import Tasks from './components/Tasks';

import { Query } from 'react-apollo';
import gql from 'graphql-tag';

export const LIST_ITEMS = gql`
  query LIST_ITEMS {
    tasks {
      id
      task
    }
  }
`

class App extends Component {
  render() {
    return (
      <div className="screen">
        <nav className="bg-primary h-10 shadow">
        <div className="pb-3">
              <h1 className="display-2 text-center pt-10 font-weight-normal text-primary" >Todo</h1>
              </div>
                <Form />
                <div className="mt-5" data-test="small">
                  <Query query={LIST_ITEMS}>
                  {({data, error, loading}) => {
                    if(error) return <p>{error.message}</p>
                    if(loading) return <p>loading...</p>
                    return data.tasks.map(task => <Tasks key={task.id} task={task} />)
                  }}
                  </Query>
                </div>
        </nav>
      </div>
    );
  }
}

export default App;
