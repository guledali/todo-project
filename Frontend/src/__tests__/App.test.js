import React from 'react';
import { mount } from 'enzyme';
import wait from 'waait';
import App, { LIST_ITEMS } from '../App';
import { MockedProvider } from 'react-apollo/test-utils';
import toJson from 'enzyme-to-json';


describe('<App/>', () => {
  it('renders with data', async() => {
    const mocks = [{
      request: { query: LIST_ITEMS },
        result: {  
        "data": {
          "tasks": [
            {
              "__typename": "Todo",
              "id": "cjp43njwfybxt0a03dsjcba8w",
              "task": "drinking"
            },
            {
              "__typename": "Todo",
              "id": "cjp43nqesybzw0a03bv91il7k",
              "task": "sleeping"
            },
          ]
        } 
      }
    }];
  
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <App />
      </MockedProvider>
    );
    await wait()
    wrapper.update()
      // console.log(wrapper.debug())
      const task = wrapper.find('[data-test="task"]');
      console.log(task.debug());
      expect(task.contains('drinking')).toEqual(true);
      expect(task.contains('sleeping')).toEqual(true);
      expect(toJson(task)).toMatchSnapshot();
  });

  it('loading data', () => {
    const mocks = [{
      request: { query: LIST_ITEMS },
        result: {  
        "data": {
          "tasks": [
            {
              "__typename": "Todo",
              "id": "cjp43njwfybxt0a03dsjcba8w",
              "task": "drinking"
            },
            {
              "__typename": "Todo",
              "id": "cjp43nqesybzw0a03bv91il7k",
              "task": "sleeping"
            },
          ]
        } 
      }
    }];
  
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <App />
      </MockedProvider>
    );
    wait()
    wrapper.update()
      console.log(wrapper.debug());
      expect(wrapper.text()).toContain("loading...");
      const loading = wrapper.find('[data-test="small"]');
      expect(toJson(loading)).toMatchSnapshot();
  });

  it('display errros', async() => {
    const mocks = [{
      request: { query: LIST_ITEMS },
      result: {
        errors: [{ message: 'Items Not Found!' }],
      },
    }];
  
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <App />
      </MockedProvider>
    );
    await wait()
    wrapper.update()
    console.log(wrapper.debug());
    expect(wrapper.text()).toContain("GraphQL error: Items Not Found!");
    const error = wrapper.find('[data-test="small"]');
    expect(toJson(error)).toMatchSnapshot();
  });
})