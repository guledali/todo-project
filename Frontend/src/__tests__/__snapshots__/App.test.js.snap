// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> display errros 1`] = `
<div
  className="mt-5"
  data-test="small"
>
  <Query
    query={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": Object {
              "kind": "Name",
              "value": "LIST_ITEMS",
            },
            "operation": "query",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "tasks",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "task",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 64,
          "start": 0,
        },
      }
    }
  >
    <p>
      GraphQL error: Items Not Found!
    </p>
  </Query>
</div>
`;

exports[`<App/> loading data 1`] = `
<div
  className="mt-5"
  data-test="small"
>
  <Query
    query={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": Object {
              "kind": "Name",
              "value": "LIST_ITEMS",
            },
            "operation": "query",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "tasks",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "task",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 64,
          "start": 0,
        },
      }
    }
  >
    <p>
      loading...
    </p>
  </Query>
</div>
`;

exports[`<App/> renders with data 1`] = `
Array [
  <span
    className="lead font-weight-normal"
    data-test="task"
  >
    drinking
  </span>,
  <span
    className="lead font-weight-normal"
    data-test="task"
  >
    sleeping
  </span>,
]
`;
